jscomp/ml/typedecl.ml

```diff
@@ -1998,8 +1998,10 @@ let report_error ppf = function
       fprintf ppf "A type parameter occurs several times"
   | Duplicate_constructor s ->
       fprintf ppf "Two constructors are named %s" s
-  | Duplicate_label s ->
-      fprintf ppf "Two labels are named %s" s
+  | Duplicate_label (s, None) ->
+      fprintf ppf "The field @{<info>%s@} is defined several times in this record. Fields can only be added once to a record." s
+  | Duplicate_label (s, Some recordName) ->
+    fprintf ppf "The field @{<info>%s@} is defined several times in the record @{<info>%s@}. Fields can only be added once to a record." s recordName
```

Comment:
I suppose the error message might be used for objects as well.

Reply:
Objects are a bit trickier, and needs a general overhaul. Doesn't seem like overwriting object fields is an error at all, although you can't change their type of course.